File: ListViewScrollExample/ListViewScrollExample/Form1.vb
Public Class Form1

    Dim vScrollHelper As Guna.UI.Lib.ScrollBar.ListViewScrollHelper
    Private Sub Form1_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        vScrollHelper = New Guna.UI.Lib.ScrollBar.ListViewScrollHelper(ListView1, GunaVScrollBar1, True)
        Dim r As New Random
        For i As Integer = 0 To 40
            Dim l As ListViewItem = New ListViewItem
            l.Text = i
            l.SubItems.Add(r.Next(200000, 500000))
            l.SubItems.Add(r.Next(200000, 500000))
            l.SubItems.Add(r.Next(200000, 500000))
            l.SubItems.Add(r.Next(200000, 500000))
            l.SubItems.Add(r.Next(200000, 500000))
            ListView1.Items.Add(l)
        Next
        vScrollHelper.UpdateScrollBar()
    End Sub

    Private Sub ListView1_Resize(sender As Object, e As EventArgs) Handles ListView1.Resize
        If vScrollHelper IsNot Nothing Then vScrollHelper.UpdateScrollBar()
    End Sub
End Class

File: ListViewScrollExample/ListViewScrollExample_CSharp/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace ListViewScrollExample_CSharp
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private Guna.UI.Lib.ScrollBar.ListViewScrollHelper vScrollHelper;
        private void Form1_Load(object sender, EventArgs e)
        {
            vScrollHelper = new Guna.UI.Lib.ScrollBar.ListViewScrollHelper(ListView1, GunaVScrollBar1, true);
            Random r = new Random();
            for (int i = 0; i <= 40; i++)
            {
                ListViewItem l = new ListViewItem();
                l.Text = i.ToString();
                l.SubItems.Add(r.Next(200000, 500000).ToString());
                l.SubItems.Add(r.Next(200000, 500000).ToString());
                l.SubItems.Add(r.Next(200000, 500000).ToString());
                l.SubItems.Add(r.Next(200000, 500000).ToString());
                l.SubItems.Add(r.Next(200000, 500000).ToString());
                ListView1.Items.Add(l);
            }
            vScrollHelper.UpdateScrollBar();
        }

        private void ListView1_Resize(object sender, EventArgs e)
        {
            if (vScrollHelper != null) vScrollHelper.UpdateScrollBar();
        }
    }
}
